Adding Upshift Pools

krishnafractol opened the pull request:
https://defillama.com/protocol/upshift

llamatester commented:
The upshift adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       52 passed, 52 total
    Snapshots:   0 total
    Time:        0.273 s
    Ran all test suites.
    
    Nb of pools: 8
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬────────────┬───────────┬───────────┬────────────────────┬──────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │ pool                                                  │ chain      │ project   │ symbol    │ tvlUsd             │ apyBase              │ underlyingTokens                                 │
    ├─────────┼───────────────────────────────────────────────────────┼────────────┼───────────┼───────────┼────────────────────┼──────────────────────┼──────────────────────────────────────────────────┤
    │ 0       │ '0x4e2D90f0307A93b54ACA31dc606F93FE6b9132d2-Base'     │ 'Base'     │ 'upshift' │ 'upLBTC'  │ 39691074.28439488  │ 1365313.5106790385   │ [ '0xecAc9C5F704e954931349Da37F60E39f515c11c1' ] │
    │ 1       │ '0xd684AF965b1c17D628ee0d77cae94259c41260F4-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upsUSDe' │ 34687120.914336294 │ 0.010073753452592978 │ [ '0x9D39A5DE30e57443BfF2A8307A4256c8797A3497' ] │
    │ 2       │ '0x5Fde59415625401278c4d41C6beFCe3790eb357f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'gtETH'   │ 21515700.3525721   │ 0.00997402195947323  │ [ '0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0' ] │
    │ 3       │ '0xEBac5e50003d4B17Be422ff9775043cD61002f7f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upBTC'   │ 13748684.852808218 │ 4.280445202147641    │ [ '0xcbB7C0000aB88B473b1f5aFd9ef808440eed33Bf' ] │
    │ 4       │ '0x18a5a3D575F34e5eBa92ac99B0976dBe26f9F869-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upLBTC'  │ 11518335.442179061 │ 0.009999928223242468 │ [ '0x8236a87084f8B84306f72007F36F2618A5634494' ] │
    │ 5       │ '0xB7858b66dFA38b9Cb74d00421316116A7851c273-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSD'   │ 8953633.65068      │ 14.550625358321069   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │
    │ 6       │ '0x3408b22d8895753C9A3e14e4222E981d4E9A599E-Avax'     │ 'Avax'     │ 'upshift' │ 'upAUSD'  │ 7406444.767904593  │ 18474928.44947099    │ [ '0x00000000eFE302BEAA2b3e6e1b18d08D69a9012a' ] │
    │ 7       │ '0x80E1048eDE66ec4c364b4F22C8768fc657FF6A42-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSDC'  │ 4080521.455432     │ 14.984456218909495   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │
    └─────────┴───────────────────────────────────────────────────────┴────────────┴───────────┴───────────┴────────────────────┴──────────────────────┴──────────────────────────────────────────────────┘

llamatester commented:
The upshift adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       52 passed, 52 total
    Snapshots:   0 total
    Time:        0.279 s
    Ran all test suites.
    
    Nb of pools: 8
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬────────────┬───────────┬───────────┬────────────────────┬──────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │ pool                                                  │ chain      │ project   │ symbol    │ tvlUsd             │ apyBase              │ underlyingTokens                                 │
    ├─────────┼───────────────────────────────────────────────────────┼────────────┼───────────┼───────────┼────────────────────┼──────────────────────┼──────────────────────────────────────────────────┤
    │ 0       │ '0x4e2D90f0307A93b54ACA31dc606F93FE6b9132d2-Base'     │ 'Base'     │ 'upshift' │ 'upLBTC'  │ 40560977.34654     │ 1365444.6274492561   │ [ '0xecAc9C5F704e954931349Da37F60E39f515c11c1' ] │
    │ 1       │ '0xd684AF965b1c17D628ee0d77cae94259c41260F4-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upsUSDe' │ 34463940.45045284  │ 0.010073753452592978 │ [ '0x9D39A5DE30e57443BfF2A8307A4256c8797A3497' ] │
    │ 2       │ '0x5Fde59415625401278c4d41C6beFCe3790eb357f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'gtETH'   │ 21171559.474315036 │ 0.009974310300390947 │ [ '0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0' ] │
    │ 3       │ '0xEBac5e50003d4B17Be422ff9775043cD61002f7f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upBTC'   │ 14109231.851767411 │ 4.280445202147641    │ [ '0xcbB7C0000aB88B473b1f5aFd9ef808440eed33Bf' ] │
    │ 4       │ '0x18a5a3D575F34e5eBa92ac99B0976dBe26f9F869-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upLBTC'  │ 11913935.47072162  │ 0.009999928223242468 │ [ '0x8236a87084f8B84306f72007F36F2618A5634494' ] │
    │ 5       │ '0xB7858b66dFA38b9Cb74d00421316116A7851c273-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSD'   │ 8958939.340252435  │ 16.707315947336603   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │
    │ 6       │ '0x3408b22d8895753C9A3e14e4222E981d4E9A599E-Avax'     │ 'Avax'     │ 'upshift' │ 'upAUSD'  │ 7455390.864172794  │ 18475695.0985991     │ [ '0x00000000eFE302BEAA2b3e6e1b18d08D69a9012a' ] │
    │ 7       │ '0x80E1048eDE66ec4c364b4F22C8768fc657FF6A42-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSDC'  │ 4091699.0060596615 │ 9.90593579543306     │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │
    └─────────┴───────────────────────────────────────────────────────┴────────────┴───────────┴───────────┴────────────────────┴──────────────────────┴──────────────────────────────────────────────────┘

slasher125 commented:
@krishnafractol apy values for some of these look v wrong, `1365444.6274492561` % for upLBTC on base?

krishnafractol commented:
> @krishnafractol apy values for some of these look v wrong, `1365444.6274492561` % for upLBTC on base?

Let me verify

llamatester commented:
The upshift adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       52 passed, 52 total
    Snapshots:   0 total
    Time:        0.282 s
    Ran all test suites.
    
    Nb of pools: 8
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬────────────┬───────────┬───────────┬────────────────────┬──────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │ pool                                                  │ chain      │ project   │ symbol    │ tvlUsd             │ apyBase              │ underlyingTokens                                 │
    ├─────────┼───────────────────────────────────────────────────────┼────────────┼───────────┼───────────┼────────────────────┼──────────────────────┼──────────────────────────────────────────────────┤
    │ 0       │ '0x4e2D90f0307A93b54ACA31dc606F93FE6b9132d2-Base'     │ 'Base'     │ 'upshift' │ 'upLBTC'  │ 38679802.5636645   │ 1365587.7389913264   │ [ '0xecAc9C5F704e954931349Da37F60E39f515c11c1' ] │
    │ 1       │ '0xd684AF965b1c17D628ee0d77cae94259c41260F4-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upsUSDe' │ 34733997.5413562   │ 0.010073753452592978 │ [ '0x9D39A5DE30e57443BfF2A8307A4256c8797A3497' ] │
    │ 2       │ '0x5Fde59415625401278c4d41C6beFCe3790eb357f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'gtETH'   │ 22479530.98843735  │ 2.276476164263822    │ [ '0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0' ] │
    │ 3       │ '0xEBac5e50003d4B17Be422ff9775043cD61002f7f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upBTC'   │ 13377292.351728749 │ 4.283736926285656    │ [ '0xcbB7C0000aB88B473b1f5aFd9ef808440eed33Bf' ] │
    │ 4       │ '0x18a5a3D575F34e5eBa92ac99B0976dBe26f9F869-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upLBTC'  │ 11221180.5583171   │ 5.29998373104407     │ [ '0x8236a87084f8B84306f72007F36F2618A5634494' ] │
    │ 5       │ '0x3408b22d8895753C9A3e14e4222E981d4E9A599E-Avax'     │ 'Avax'     │ 'upshift' │ 'upAUSD'  │ 7412972.968076     │ 18516034.85410289    │ [ '0x00000000eFE302BEAA2b3e6e1b18d08D69a9012a' ] │
    │ 6       │ '0x80E1048eDE66ec4c364b4F22C8768fc657FF6A42-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSDC'  │ 7092616.96697204   │ 13.38392646616159    │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │
    │ 7       │ '0xB7858b66dFA38b9Cb74d00421316116A7851c273-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSD'   │ 6203494.046892166  │ 16.58383733743218    │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │
    └─────────┴───────────────────────────────────────────────────────┴────────────┴───────────┴───────────┴────────────────────┴──────────────────────┴──────────────────────────────────────────────────┘

krishnafractol commented:
> slasher125

Hi @slasher125 I've fixed the calculation APR looks correct now please review

slasher125 commented:
> > slasher125
> 
> Hi @slasher125 I've fixed the calculation APR looks correct now please review

great. pls revert the change in package-lock

krishnafractol commented:
> > > slasher125
> > 
> > 
> > Hi @slasher125 I've fixed the calculation APR looks correct now please review
> 
> great. pls revert the change in package-lock

Removed it

llamatester commented:
The upshift adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       53 passed, 53 total
    Snapshots:   0 total
    Time:        0.287 s
    Ran all test suites.
    
    Nb of pools: 8
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬────────────┬───────────┬───────────┬────────────────────┬──────────────────────┬──────────────────────────────────────────────────┬────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │ pool                                                  │ chain      │ project   │ symbol    │ tvlUsd             │ apyBase              │ underlyingTokens                                 │ apyReward          │ rewardTokens                                     │
    ├─────────┼───────────────────────────────────────────────────────┼────────────┼───────────┼───────────┼────────────────────┼──────────────────────┼──────────────────────────────────────────────────┼────────────────────┼──────────────────────────────────────────────────┤
    │ 0       │ '0x4e2D90f0307A93b54ACA31dc606F93FE6b9132d2-Base'     │ 'Base'     │ 'upshift' │ 'upLBTC'  │ 39962135.85919503  │ 3.190472925531595    │ [ '0xecAc9C5F704e954931349Da37F60E39f515c11c1' ] │                    │                                                  │
    │ 1       │ '0xd684AF965b1c17D628ee0d77cae94259c41260F4-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upsUSDe' │ 35036032.302585386 │ 0.010073753452592978 │ [ '0x9D39A5DE30e57443BfF2A8307A4256c8797A3497' ] │                    │                                                  │
    │ 2       │ '0x5Fde59415625401278c4d41C6beFCe3790eb357f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'gtETH'   │ 23127427.281430576 │ 2.276476164263822    │ [ '0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0' ] │                    │                                                  │
    │ 3       │ '0xEBac5e50003d4B17Be422ff9775043cD61002f7f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upBTC'   │ 13816010.356695    │ 4.283736926285656    │ [ '0xcbB7C0000aB88B473b1f5aFd9ef808440eed33Bf' ] │                    │                                                  │
    │ 4       │ '0x18a5a3D575F34e5eBa92ac99B0976dBe26f9F869-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upLBTC'  │ 11587373.187261691 │ 5.29998373104407     │ [ '0x8236a87084f8B84306f72007F36F2618A5634494' ] │                    │                                                  │
    │ 5       │ '0xB7858b66dFA38b9Cb74d00421316116A7851c273-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSD'   │ 8304048.025422333  │ 12.390005643116076   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │                    │                                                  │
    │ 6       │ '0x3408b22d8895753C9A3e14e4222E981d4E9A599E-Avax'     │ 'Avax'     │ 'upshift' │ 'upAUSD'  │ 7416250.491701     │ 15.81149177184708    │ [ '0x00000000eFE302BEAA2b3e6e1b18d08D69a9012a' ] │ 14.873698863347029 │ [ '0x0000000000000000000000000000000000000000' ] │
    │ 7       │ '0x80E1048eDE66ec4c364b4F22C8768fc657FF6A42-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSDC'  │ 7101709.978617658  │ 13.631031098219403   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │                    │                                                  │
    └─────────┴───────────────────────────────────────────────────────┴────────────┴───────────┴───────────┴────────────────────┴──────────────────────┴──────────────────────────────────────────────────┴────────────────────┴──────────────────────────────────────────────────┘

llamatester commented:
The upshift adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       53 passed, 53 total
    Snapshots:   0 total
    Time:        0.297 s
    Ran all test suites.
    
    Nb of pools: 8
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬────────────┬───────────┬───────────┬────────────────────┬──────────────────────┬──────────────────────────────────────────────────┬────────────────────┬──────────────────────────────────────────────────┐
    │ (index) │ pool                                                  │ chain      │ project   │ symbol    │ tvlUsd             │ apyBase              │ underlyingTokens                                 │ apyReward          │ rewardTokens                                     │
    ├─────────┼───────────────────────────────────────────────────────┼────────────┼───────────┼───────────┼────────────────────┼──────────────────────┼──────────────────────────────────────────────────┼────────────────────┼──────────────────────────────────────────────────┤
    │ 0       │ '0x4e2D90f0307A93b54ACA31dc606F93FE6b9132d2-Base'     │ 'Base'     │ 'upshift' │ 'upLBTC'  │ 39962135.85919503  │ 3.190472925531595    │ [ '0xecAc9C5F704e954931349Da37F60E39f515c11c1' ] │                    │                                                  │
    │ 1       │ '0xd684AF965b1c17D628ee0d77cae94259c41260F4-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upsUSDe' │ 35036032.302585386 │ 0.010073753452592978 │ [ '0x9D39A5DE30e57443BfF2A8307A4256c8797A3497' ] │                    │                                                  │
    │ 2       │ '0x5Fde59415625401278c4d41C6beFCe3790eb357f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'gtETH'   │ 23127427.281430576 │ 2.276476164263822    │ [ '0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0' ] │                    │                                                  │
    │ 3       │ '0xEBac5e50003d4B17Be422ff9775043cD61002f7f-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upBTC'   │ 13818502.32742575  │ 4.283736926285656    │ [ '0xcbB7C0000aB88B473b1f5aFd9ef808440eed33Bf' ] │                    │                                                  │
    │ 4       │ '0x18a5a3D575F34e5eBa92ac99B0976dBe26f9F869-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upLBTC'  │ 11587373.187261691 │ 5.29998373104407     │ [ '0x8236a87084f8B84306f72007F36F2618A5634494' ] │                    │                                                  │
    │ 5       │ '0xB7858b66dFA38b9Cb74d00421316116A7851c273-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSD'   │ 8304023.113078958  │ 12.390005643116076   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │                    │                                                  │
    │ 6       │ '0x3408b22d8895753C9A3e14e4222E981d4E9A599E-Avax'     │ 'Avax'     │ 'upshift' │ 'upAUSD'  │ 7416250.491701     │ 15.81149177184708    │ [ '0x00000000eFE302BEAA2b3e6e1b18d08D69a9012a' ] │ 14.869290535290316 │ [ '0x0000000000000000000000000000000000000000' ] │
    │ 7       │ '0x80E1048eDE66ec4c364b4F22C8768fc657FF6A42-Ethereum' │ 'Ethereum' │ 'upshift' │ 'upUSDC'  │ 7101688.67331728   │ 13.631031098219403   │ [ '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' ] │                    │                                                  │
    └─────────┴───────────────────────────────────────────────────────┴────────────┴───────────┴───────────┴────────────────────┴──────────────────────┴──────────────────────────────────────────────────┴────────────────────┴──────────────────────────────────────────────────┘
